sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -497,6 +498,41 @@ class Analyzer(
           ordering.map(order => resolveExpression(order, child).asInstanceOf[SortOrder])
         Sort(newOrdering, global, child)
 
+      // Replace the index with the related attribute for ORDER BY
+      // which is a 1-base position of the projection list.
+      case s @ Sort(orders, global, child) if child.resolved &&
```

Comment:
I am unable to find a good place for `group by ordinal resolution`, after placing `order by ordinal resolution` in `ResolveSortReferences`. Two options are in my mind:

- Assuming we can merge https://github.com/apache/spark/pull/11208, which splits `ResolveReferences` to two rules: `ResolveStar` and `ResolveReferences`. Then, `ResolveReferences` is not very long, maybe we can keep resolution of ordinal here.
- Create a separate rule `ResolveOrdinal` for both cases.

In the next PR, I will first pick the second option, if nobody is against it. : ) Thanks!